### PR TITLE
Max requests param for gunicorn in docker

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

Adds a max requests parameter for gunicorn in the server Docker image. This would allow reloading the worker after a some requests in case it has banked up a lot of memory allocation.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by CR
- [x] tests executed by DN

### CODE REVIEW

**Comment by dnil** [comment]
✅ Parameter is passed to workers on stage:
Screenshot 2023-10-19 at 19 09 50

**Comment by dnil** [comment]
Still unsure about the exact number. Not 1. Not 10000.

File: Dockerfile-server [hunk 1]
```diff
@@ -57,12 +57,14 @@ ENV GUNICORN_WORKERS=1
 ENV GUNICORN_THREADS=1
 ENV GUNICORN_BIND="0.0.0.0:8000"
 ENV GUNICORN_TIMEOUT=400
+ENV GUNICORN_MAXREQUESTS=2000
```
Comment (northwestwitch) [inline Dockerfile-server:60]: How did you end up with 2000?

Comment (dnil) [reply Dockerfile-server:60]: Ballparking! Feel very very free to suggest somthing else. Im testing around a bit on stage, but as we know we can't quite replicate, so...

Comment (dnil) [reply Dockerfile-server:60]: Honestly, I feel we are giving mem back quite a bit, its not just explosive linear growth, though it is certainly growth over time.

Comment (northwestwitch) [reply Dockerfile-server:60]: I don't have any better number in mind, so 2000 is fine by me, let's hope it works!

**Comment by northwestwitch** [comment]
> Still unsure about the exact number. Not 1. Not 10000.

I've asked ChatGPT, they say: `A common practice is to set --max-requests to a reasonable value, such as 1000 or 10000, and then adjust it based on your application's behavior and server performance. Regularly monitor your server, and if you notice any issues or resource leaks, consider lowering the value.`

**Comment by dnil** [comment]
Any wiser? We can perhaps say we already have issues, and so start lower, like 500?

**Comment by northwestwitch** [comment]
> Any wiser? We can perhaps say we already have issues, and so start lower, like 500?

Ok, I've asked again
Q: My flask app has a memory leak. I'm considering reducing the max gunicorn requests, which number of max requests is reccommended?

A: Experiment: Set an initial max_requests value that you think is reasonable, and monitor how your application behaves. You might start with a value like 100 or 1000, but the actual number will depend on your specific circumstances.

**Comment by dnil** [comment]
Maybe it is googling this conversation? 😉

**Comment by northwestwitch** [comment]
> Maybe it is googling this conversation? 😉

It's probably going to learn from our mistakes and the next one will have the solution

**Review by northwestwitch — APPROVED** [review]
🤞🏻 💯

**Pull request merged**